File: src/com/suushiemaniac/cubing/bld/model/enumeration/piece/PieceType.java
package com.suushiemaniac.cubing.bld.model.enumeration.piece;

import com.suushiemaniac.cubing.alglib.lang.NotationReader;

public interface PieceType {
	int getTargetsPerPiece();

	int getNumPieces();

	default int getNumPiecesNoBuffer() {
		return this.getNumPieces() - 1;
	}

	int getNumAlgs();

	String name();

	String humanName();

	String mnemonic();

	NotationReader getReader();
}

File: src/com/suushiemaniac/cubing/bld/model/enumeration/piece/PieceType.kt
package com.suushiemaniac.cubing.bld.model.enumeration.piece

import com.suushiemaniac.cubing.alglib.lang.NotationReader

interface PieceType {
    val targetsPerPiece: Int

    val numPieces: Int

    val numPiecesNoBuffer: Int
        get() = this.numPieces - 1

    val numAlgs: Int

    val reader: NotationReader

    val name: String

    val humanName: String

    val mnemonic: String
}
